tests/c/side-trace.c

```diff
@@ -0,0 +1,103 @@
+// Run-time:
+//   env-var: YKD_PRINT_IR=jit-pre-opt
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_PRINT_JITSTATE=1
+//   stderr:
+//     jit-state: start-tracing
+//     1
+//     jit-state: stop-tracing
+//     --- Begin jit-pre-opt ---
+//     ...
+//     --- End jit-pre-opt ---
+//     2
+//     jit-state: enter-jit-code
+//     3
+//     4
+//     5
+//     6
+//     7
+//     8
+//     9
+//     10
+//     jit-state: deoptimise
+//     12
+//     jit-state: enter-jit-code
+//     jit-state: deoptimise
+//     14
+//     jit-state: enter-jit-code
+//     jit-state: deoptimise
+//     16
+//     jit-state: enter-jit-code
+//     jit-state: deoptimise
+//     18
+//     jit-state: enter-jit-code
+//     jit-state: deoptimise
+//     jit-state: start-side-tracing
+//     20
+//     jit-state: stop-side-tracing
+//     --- Begin jit-pre-opt ---
+//     ...
+//     --- End jit-pre-opt ---
+//     22
+//     jit-state: enter-jit-code
+//     jit-state: execute-side-trace
+//     24
+//     jit-state: deoptimise
+//     26
+//     jit-state: enter-jit-code
+//     jit-state: execute-side-trace
+//     28
+//     jit-state: deoptimise
+//     30
+//   stdout:
+//     exit
+
+// Testing side tracing functionality. The program immediately compiles a
+// trace, which runs for 10 iterations at which point the guard generated by
+// the if/else in function `foo` fails. After failing for 5 times, a side trace
```

Comment:
Says after 5 times here, but you said 50 elsewhere?

Reply:
50 is the default, but for this test I changed it to 5. See https://github.com/ykjit/yk/pull/828/files/3b64ba16a80c11718bf1ea730eb756982d48cd7d#diff-66116c893a3e9b6083a5eefd36f66447a0461fd33defc495275221ad16457e77R84